### PR TITLE
F/wep 78 editor multi language

### PR DESCRIPTION
**Pull request by 0815pascal**

### CODE REVIEW

File: packages/editor/src/client/base.tsx [hunk 1]
```diff
@@ -29,55 +29,59 @@ import {
...
-            label="Articles"
+            label={t('Articles')}
             route={ArticleListRoute.create({})}
             active={current?.type === RouteType.ArticleList}
           />
 
           <RouteMenuLinkButton
             icon={MaterialIconInsertDriveFileOutlined}
-            label="Pages"
+            label={t('Some Pages')}
```
Comment (roosnic1) [inline packages/editor/src/client/base.tsx:56]: some pages?

File: packages/editor/src/client/blocks/teaserGridBlock.tsx [hunk 2]
```diff
@@ -332,12 +338,12 @@ export function TeaserContent({
         <Box display="flex" flexWrap="wrap">
           <Box flexShrink={0} marginRight={Spacing.ExtraSmall}>
             <Typography variant="subtitle1" color="gray">
-              Style: {labelForTeaserStyle(style)}
+              {t('Style')}: {labelForTeaserStyle(style)}
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/teaserGridBlock.tsx:341]: You should use interpolation for those translations. https://www.i18next.com/translation-function/interpolation

File: packages/editor/src/client/blocks/teaserGridBlock.tsx [hunk 3]
```diff
@@ -332,12 +338,12 @@ export function TeaserContent({
...
             </Typography>
           </Box>
           <Box flexShrink={0}>
             <Typography variant="subtitle1" color="gray">
-              Status: {states?.join(' / ')}
+              {t('Status')}: {states?.join(' / ')}
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/teaserGridBlock.tsx:346]: You should use interpolation for those translations. https://www.i18next.com/translation-function/interpolation

File: packages/editor/src/client/panel/embedEditPanel.tsx [hunk 4]
```diff
@@ -114,29 +117,33 @@ export function EmbedEditPanel({value, onClose, onConfirm}: EmbedEditPanel) {
...
           <code>{'<iframe width="560" height="315" src="https://..."></iframe>'}</code>
           <Typography variant="subtitle1" spacing="small">
-            Set <code>style</code> alternatively as attribute to overwrite the defaults and disable
-            auto ratio size scaling .e.g.
+            {t(
+              'Set <code>style</code> alternatively as attribute to overwrite the defaults and disable auto ratio size scaling .e.g.'
+            )}
           </Typography>
+
           <code>{'<iframe style="height:350px;width:100%;" src="https://..."></iframe>'}</code>
```
Comment (roosnic1) [inline packages/editor/src/client/panel/embedEditPanel.tsx:142]: Why didn't you put this string into the translations?

File: packages/editor/src/client/panel/embedEditPanel.tsx [hunk 5]
```diff
@@ -114,29 +117,33 @@ export function EmbedEditPanel({value, onClose, onConfirm}: EmbedEditPanel) {
...
           errorMessage={errorMessage}
           value={input}
           onChange={e => setInput(e.target.value)}
         />
         <Box marginBottom={Spacing.ExtraSmall}>
           <Typography variant="subtitle1" spacing="small">
-            Facebook Post/Video, Instagram Post, Twitter Tweet, Vimeo Video, YouTube Video formatted
-            like e.g.
+            {t(
+              'Facebook Post/Video, Instagram Post, Twitter Tweet, Vimeo Video, YouTube Video formatted like e.g.'
```
Comment (roosnic1) [inline packages/editor/src/client/panel/embedEditPanel.tsx:128]: It's not really useful to use such a long key.

File: packages/editor/src/client/panel/embedEditPanel.tsx [hunk 6]
```diff
@@ -114,29 +117,33 @@ export function EmbedEditPanel({value, onClose, onConfirm}: EmbedEditPanel) {
...
+            )}
           </Typography>
           <code>https://www.facebook.com/id/posts/id/</code>
           <Typography variant="subtitle1" spacing="small">
-            Embed codes attributes <code>title, src, width, height</code> are validated e.g.
+            {t('Embed codes attributes <code>title, src, width, height</code> are validated e.g.')}
```
Comment (roosnic1) [inline packages/editor/src/client/panel/embedEditPanel.tsx:133]: It's not really useful to use such a long key.

File: packages/editor/src/client/panel/embedEditPanel.tsx [hunk 7]
```diff
@@ -114,29 +117,33 @@ export function EmbedEditPanel({value, onClose, onConfirm}: EmbedEditPanel) {
...
           </Typography>
           <code>{'<iframe width="560" height="315" src="https://..."></iframe>'}</code>
           <Typography variant="subtitle1" spacing="small">
-            Set <code>style</code> alternatively as attribute to overwrite the defaults and disable
-            auto ratio size scaling .e.g.
+            {t(
+              'Set <code>style</code> alternatively as attribute to overwrite the defaults and disable auto ratio size scaling .e.g.'
```
Comment (roosnic1) [inline packages/editor/src/client/panel/embedEditPanel.tsx:138]: It's not really useful to use such a long key.

File: packages/editor/src/client/panel/embedEditPanel.tsx [hunk 8]
```diff
@@ -114,29 +117,33 @@ export function EmbedEditPanel({value, onClose, onConfirm}: EmbedEditPanel) {
...
+            )}
           </Typography>
+
           <code>{'<iframe style="height:350px;width:100%;" src="https://..."></iframe>'}</code>
           <Typography variant="subtitle1" spacing="small">
-            Due to validation, shareable peers and GDPR-compliant, embedding blocks are currently
-            limited to simple iframes and supported embeds listed above.
+            {t(
+              'Due to validation, shareable peers and GDPR-compliant, embedding blocks are currently limited to simple iframes and supported embeds listed above.'
```
Comment (roosnic1) [inline packages/editor/src/client/panel/embedEditPanel.tsx:145]: It's not really useful to use such a long key.

File: packages/editor/src/client/panel/publishArticlePanel.tsx [hunk 9]
```diff
@@ -73,15 +77,15 @@ export function PublishArticlePanel({
         {pendingPublishDate && (
           <Box marginBottom={Spacing.Small}>
             <Typography variant="subtitle1" color="alert">
-              There is already a pending publication scheduled at{' '}
+              {t('There is already a pending publication scheduled at')}{' '}
```
Comment (roosnic1) [inline packages/editor/src/client/panel/publishArticlePanel.tsx:80]: You should use formatting for this whole Typography block => https://www.i18next.com/translation-function/formatting

File: packages/editor/src/client/panel/publishPagePanel.tsx [hunk 10]
```diff
@@ -72,15 +76,15 @@ export function PublishPagePanel({
         {pendingPublishDate && (
           <Box marginBottom={Spacing.Small}>
             <Typography variant="subtitle1" color="alert">
-              There is already a pending publication scheduled at{' '}
+              {t('There is already a pending publication scheduled at')}{' '}
```
Comment (roosnic1) [inline packages/editor/src/client/panel/publishPagePanel.tsx:79]: You should use formatting for this whole Typography block => https://www.i18next.com/translation-function/formatting

File: packages/editor/src/client/panel/resetUserPasswordPanel.tsx [hunk 11]
```diff
@@ -35,29 +37,37 @@ export function ResetUserPasswordPanel({userID, userName, onClose}: ResetUserPas
...
+              icon={MaterialIconClose}
+              label={t('Close')}
+              onClick={() => onClose()}
+            />
           }
         />
         <PanelSection>
-          <Typography variant="body1">Reset the password for {userName}</Typography>
+          <Typography variant="body1">
+            {t('Reset the password for')} {userName}
```
Comment (roosnic1) [inline packages/editor/src/client/panel/resetUserPasswordPanel.tsx:57]: https://www.i18next.com/translation-function/interpolation

File: packages/editor/src/client/panel/tokenGeneratePanel.tsx [hunk 12]
```diff
@@ -72,8 +80,9 @@ export function TokenGeneratePanel({onClose}: TokenGeneratePanelProps) {
             <>
               <Box marginBottom={Spacing.Small}>
                 <Typography variant="body1">
-                  Successfully created token, make sure to copy it now. You won’t be able to see it
-                  again!
+                  {t(
+                    'Successfully created token, make sure to copy it now. You won’t be able to see it again!'
```
Comment (roosnic1) [inline packages/editor/src/client/panel/tokenGeneratePanel.tsx:84]: too long

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: packages/editor/src/client/base.tsx (hunk 1 above)
Comment (0815pascal) [reply packages/editor/src/client/base.tsx:56]: I applied that from the template you made; I thought this was for a particular reason that you named it "some pages" :) Switched it back to "Pages" now

File: packages/editor/src/client/panel/embedEditPanel.tsx (hunk 4 above)
Comment (0815pascal) [reply packages/editor/src/client/panel/embedEditPanel.tsx:142]: Because it's pure code and there's nothing to translate. I thought it wouldn't make sense to populate the dictionary with this

File: packages/editor/src/client/panel/embedEditPanel.tsx (hunk 5 above)
Comment (0815pascal) [reply packages/editor/src/client/panel/embedEditPanel.tsx:128]: Done

File: packages/editor/src/client/panel/embedEditPanel.tsx (hunk 8 above)
Comment (0815pascal) [reply packages/editor/src/client/panel/embedEditPanel.tsx:145]: Done

File: packages/editor/src/client/panel/tokenGeneratePanel.tsx (hunk 12 above)
Comment (0815pascal) [reply packages/editor/src/client/panel/tokenGeneratePanel.tsx:84]: Done

File: packages/editor/src/client/panel/embedEditPanel.tsx (hunk 7 above)
Comment (0815pascal) [reply packages/editor/src/client/panel/embedEditPanel.tsx:138]: Done

File: packages/editor/src/client/panel/embedEditPanel.tsx (hunk 6 above)
Comment (0815pascal) [reply packages/editor/src/client/panel/embedEditPanel.tsx:133]: Done

File: packages/editor/src/client/blocks/teaserGridBlock.tsx (hunk 3 above)
Comment (0815pascal) [reply packages/editor/src/client/blocks/teaserGridBlock.tsx:346]: Done

File: packages/editor/src/client/panel/resetUserPasswordPanel.tsx (hunk 11 above)
Comment (0815pascal) [reply packages/editor/src/client/panel/resetUserPasswordPanel.tsx:57]: Done

File: packages/editor/src/client/blocks/teaserGridBlock.tsx (hunk 2 above)
Comment (0815pascal) [reply packages/editor/src/client/blocks/teaserGridBlock.tsx:341]: Done

File: packages/editor/src/client/blocks/teaserGridBlock.tsx (hunk 3 above)
Comment (0815pascal) [reply packages/editor/src/client/blocks/teaserGridBlock.tsx:346]: Done

File: packages/editor/src/client/panel/resetUserPasswordPanel.tsx (hunk 11 above)
Comment (0815pascal) [reply packages/editor/src/client/panel/resetUserPasswordPanel.tsx:57]: Done

File: packages/editor/src/client/panel/embedEditPanel.tsx (hunk 4 above)
Comment (roosnic1) [reply packages/editor/src/client/panel/embedEditPanel.tsx:142]: If it's text that is displayed to the users it should be in the translation files. The name translations is a bit confusing here.

Comment (0815pascal) [reply packages/editor/src/client/panel/embedEditPanel.tsx:142]: Done

File: packages/editor/src/client/panel/publishArticlePanel.tsx (hunk 9 above)
Comment (0815pascal) [reply packages/editor/src/client/panel/publishArticlePanel.tsx:80]: Done

File: yarn.lock [hunk 13]
```diff
@@ -18585,12 +18585,11 @@ vm-browserify@^1.0.1:
 void-elements@^2.0.1:
   version "2.0.1"
   resolved "https://registry.yarnpkg.com/void-elements/-/void-elements-2.0.1.tgz#c066afb582bb1cb4128d60ea92392e94d5e9dbec"
-  integrity sha1-wGavtYK7HLQSjWDqkjkulNXp2+w=
 
-vue-template-compiler@^2.6.11:
-  version "2.6.11"
-  resolved "https://registry.yarnpkg.com/vue-template-compiler/-/vue-template-compiler-2.6.11.tgz#c04704ef8f498b153130018993e56309d4698080"
-  integrity sha512-KIq15bvQDrcCjpGjrAhx4mUlyyHfdmTaoNfeoATHLAiWB+MU3cx4lOzMwrnUh9cCxy0Lt1T11hAFY6TQgroUAA==
+vue-template-compiler@^2.6.12:
```
Comment (roosnic1) [inline yarn.lock:18589]: This looks very wrong.

File: yarn.lock [hunk 14]
```diff
@@ -1252,7 +1252,7 @@
   dependencies:
     regenerator-runtime "^0.13.4"
 
-"@babel/runtime@^7.10.1":
+"@babel/runtime@^7.10.1", "@babel/runtime@^7.11.2":
```
Comment (roosnic1) [inline yarn.lock:1255]: why did you update @babel?

File: packages/editor/src/client/panel/publishArticlePanel.tsx [hunk 15]
```diff
@@ -73,13 +73,12 @@ export function PublishArticlePanel({
           />
         }
       />
+      {/** TODO: Insert language specific Tag here in toLocaleDateString and toLocaleTimeString */}
```
Comment (roosnic1) [inline packages/editor/src/client/panel/publishArticlePanel.tsx:76]: Is this todo still needed?

File: packages/editor/src/client/panel/pageMetadataPanel.tsx [hunk 16]
```diff
@@ -30,6 +30,11 @@ import {ImageSelectPanel} from './imageSelectPanel'
 import {ImageRefFragment} from '../api'
 
 import {useTranslation} from 'react-i18next'
+export interface PageMetadataProperty {
```
Comment (roosnic1) [inline packages/editor/src/client/panel/pageMetadataPanel.tsx:33]: Why did you add this?

File: packages/editor/src/client/panel/articleMetadataPanel.tsx [hunk 17]
```diff
@@ -39,6 +39,11 @@ import {slugify} from '../utility'
 import {useAuthorListQuery, AuthorRefFragment, ImageRefFragment} from '../api'
 
 import {useTranslation} from 'react-i18next'
+export interface ArticleMetadataProperty {
```
Comment (roosnic1) [inline packages/editor/src/client/panel/articleMetadataPanel.tsx:42]: why did you add this?

**Review by roosnic1 — CHANGES_REQUESTED** [review]
@0815pascal did you test the date formatting and interpolation?

File: packages/editor/src/client/panel/pageMetadataPanel.tsx (hunk 16 above)
Comment (0815pascal) [reply packages/editor/src/client/panel/pageMetadataPanel.tsx:33]: I merged changes in the master-branch into mine. This line was added during the merge I guess.

File: packages/editor/src/client/panel/articleMetadataPanel.tsx (hunk 17 above)
Comment (0815pascal) [reply packages/editor/src/client/panel/articleMetadataPanel.tsx:42]: Same here

File: yarn.lock (hunk 14 above)
Comment (0815pascal) [reply yarn.lock:1255]: that's also because of the merge. the master-branch uses 7.11.2 [see here](https://github.com/wepublish/wepublish/blob/2968cf711d39c58dcec51916a4803056f3528f17/yarn.lock#L1255)

File: yarn.lock (hunk 13 above)
Comment (0815pascal) [reply yarn.lock:18589]: It's in the yarn.lock [in master-branch](https://github.com/wepublish/wepublish/blob/2968cf711d39c58dcec51916a4803056f3528f17/yarn.lock#L18556)

Comment (roosnic1) [reply yarn.lock:18589]: Sorry. I got confused. I didn't know that github highlighted those changes.

File: yarn.lock (hunk 14 above)
Comment (roosnic1) [reply yarn.lock:1255]: Sorry. I got confused. I didn't know that github highlighted those changes.

File: packages/editor/src/client/panel/pageMetadataPanel.tsx (hunk 16 above)
Comment (roosnic1) [reply packages/editor/src/client/panel/pageMetadataPanel.tsx:33]: Sorry. I got confused. I didn't know that github highlighted those changes.

File: packages/editor/src/client/panel/articleMetadataPanel.tsx (hunk 17 above)
Comment (roosnic1) [reply packages/editor/src/client/panel/articleMetadataPanel.tsx:42]: Sorry. I got confused. I didn't know that github highlighted those changes.

File: packages/editor/src/client/blocks/teaserGridBlock.tsx [hunk 18]
```diff
@@ -203,13 +207,14 @@ export function TeaserBlock({
...
   switch (teaser.type) {
     case TeaserType.Article: {
       const states = []
 
-      if (teaser?.article?.draft) states.push('Draft')
-      if (teaser?.article?.pending) states.push('Pending')
-      if (teaser?.article?.published) states.push('Published')
+      if (teaser?.article?.draft) states.push(t('draft'))
+      if (teaser?.article?.pending) states.push(t('pending'))
+      if (teaser?.article?.published) states.push(t('published'))
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/teaserGridBlock.tsx:217]: Please use a more descriptive translation key for the three states. Something like `state_draft`

File: packages/editor/src/client/blocks/teaserGridBlock.tsx [hunk 19]
```diff
@@ -280,6 +285,20 @@ export interface TeaserContentProps {
...
+      return t('styleDefault')
+
+    case TeaserStyle.Light:
+      return t('styleLight')
+
+    case TeaserStyle.Text:
+      return t('styleText')
+  }
+}
+
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/teaserGridBlock.tsx:301]: This shouldn't be translated.

File: packages/editor/src/client/blocks/teaserGridBlock.tsx [hunk 20]
```diff
@@ -332,29 +351,16 @@ export function TeaserContent({
         <Box display="flex" flexWrap="wrap">
           <Box flexShrink={0} marginRight={Spacing.ExtraSmall}>
             <Typography variant="subtitle1" color="gray">
-              Style: {labelForTeaserStyle(style)}
+              {labelForTeaserStyle(style)}
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/teaserGridBlock.tsx:354]: You should use interpolation here

File: packages/editor/src/client/blocks/teaserGridBlock.tsx [hunk 21]
```diff
@@ -332,29 +351,16 @@ export function TeaserContent({
...
             </Typography>
           </Box>
           <Box flexShrink={0}>
             <Typography variant="subtitle1" color="gray">
-              Status: {states?.join(' / ')}
+              {states?.join(' / ')}
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/teaserGridBlock.tsx:359]: What happened to `Status:`? Here you can work with two translations.

File: packages/editor/src/client/panel/imageEditPanel.tsx [hunk 22]
```diff
@@ -247,85 +251,87 @@ export function ImagedEditPanel({id, file, onClose, onSave}: ImageEditPanelProps
...
               <DescriptionList>
-                <DescriptionListItem label="Filename">
+                <DescriptionListItem label={t('Filename')}>
                   {filename || 'untitled'}
                   {extension}
                 </DescriptionListItem>
-                <DescriptionListItem label="Dimension">
-                  {imageWidth} x {imageHeight}
+                <DescriptionListItem label={t('Dimension')}>
+                  {imageWidth} {t('x')} {imageHeight}
```
Comment (roosnic1) [inline packages/editor/src/client/panel/imageEditPanel.tsx:259]: interpolation

**Review by roosnic1 — CHANGES_REQUESTED** [review]
Just a few tiny changes

File: packages/editor/src/client/blocks/teaserGridBlock.tsx (hunk 21 above)
Comment (0815pascal) [reply packages/editor/src/client/blocks/teaserGridBlock.tsx:359]: I've included it directly in the translation `"state_published": "Status: Published"` - but when taking a closer look at the code now I wonder if this means that a post can have two states at the same time such as "published" and "pending" ? Then of course what I made doesn't work

**Review by penk85 — DISMISSED** [review]

**Review by roosnic1 — CHANGES_REQUESTED** [review]
@0815pascal Looks good but at the bottom of the `en.json` you have a lot of translations i'm pretty sure you don't need anymore.

**Review by roosnic1 — APPROVED** [review]
👍🏾
